app/controllers/RealTimeIncomeInformationController.scala

```diff
@@ -16,126 +16,79 @@
-    implicit request =>
-      authorised(AuthProviders(PrivilegedApplication)) {
-        if (validateCorrelationId(correlationId)) {
-          validateDates(request.body) match {
-            case Right(_) => retrieveCitizenIncome(correlationId)
-            case Left(failure: DesSingleFailureResponse) => Future.successful(BadRequest(Json.toJson(failure)))
-          }
-        } else {
-          Future.successful(BadRequest(Json.toJson(Constants.responseInvalidCorrelationId)))
-        }
-      } recover {
-        case _: UnsupportedAuthProvider => Forbidden(Json.toJson(Constants.responseNonPrivilegedApplication))
-      }
-  }
+  val logger: Logger = Logger(this.getClass)
 
-  private def retrieveCitizenIncome(correlationId: String)(implicit hc: HeaderCarrier, request: Request[JsValue]) = {
-    schemaValidationHandler(request.body) match {
-      case Left(JsError(_)) => Future.successful(BadRequest(Json.toJson(Constants.responseInvalidPayload)))
-      case Right(JsSuccess(_, _)) => withJsonBody[RequestDetails] {
-        body =>
-          auditService.rtiiAudit(correlationId, body)
-          rtiiService.retrieveCitizenIncome(body, correlationId) map {
-            case filteredResponse: DesFilteredSuccessResponse => Ok(Json.toJson(filteredResponse))
-            case noMatchResponse: DesSuccessResponse => Ok(Json.toJson(noMatchResponse))
-            case singleFailureResponse: DesSingleFailureResponse => failureResponseToResult(singleFailureResponse)
-            case multipleFailureResponse: DesMultipleFailureResponse => BadRequest(Json.toJson(multipleFailureResponse))
-            case unexpectedResponse: DesUnexpectedResponse => InternalServerError(Json.toJson(unexpectedResponse))
-          } recover {
-            case NonFatal(_) =>
-              ServiceUnavailable(Json.toJson(DesSingleFailureResponse(Constants.errorCodeServiceUnavailable,
-                "Dependent systems are currently not responding.")))
-          }
+  def preSchemaValidation(correlationId: String): Action[JsValue] = authenticateAndValidate(correlationId).async(parse.json) {
+    implicit request =>
+      (parseJson andThen validateDate andThen validateAgainstSchema(request.body) andThen getResult)(request) {
+       requestDetails =>
+         auditService.rtiiAudit(correlationId, requestDetails)
+         rtiiService.retrieveCitizenIncome(requestDetails, correlationId) map {
+           case filteredResponse: DesFilteredSuccessResponse => Ok(Json.toJson(filteredResponse))
+           case noMatchResponse: DesSuccessResponse => Ok(Json.toJson(noMatchResponse))
+           case singleFailureResponse: DesSingleFailureResponse => failureResponseToResult(singleFailureResponse)
+           case multipleFailureResponse: DesMultipleFailureResponse => BadRequest(Json.toJson(multipleFailureResponse))
+           case unexpectedResponse: DesUnexpectedResponse => InternalServerError(Json.toJson(unexpectedResponse))
```

Comment:
Changing this may cause problems for our consumers. Its fine for internal backends, but Im not sure you can send 502 off platform? @pascal22p do you know more about this?

Reply:
There was nothing in the schema about any 5XX responses

Reply:
Error codes were agreed with DWP when this service was built. I think 500's were masked to avoid multiple retries on DES and the agreement was made we would handle that ourselves as no records could be persisted and it was a case of come back later and try. @Synnnapse might be able to confirm or deny this.